# Laboratorio 7 - Parte 1

## Integrantes

- Sergio Orellana 221122
- Rodrigo Mansilla 22611
- Carlos Valladares 221164

## Parte 1 - Teoría

### Pregunta 1
**Describa con sus propias palabras la diferencia fundamental entre una Entidad de DES y un Agente de MBA. ¿Por qué esta diferencia hace que cada paradigma sea adecuado para tipos de problemas distintos? Proporcione un ejemplo de un problema de modelado para cada uno**

En DES, una *entidad* (por ejemplo, un cliente o una orden) es típicamente pasiva: fluye a través de un proceso predefinido (llegar → esperar → ser atendida → salir) y “cobra vida” solo cuando ocurre un evento del sistema (arriba una llegada, termina un servicio, cambia un recurso). La lógica dominante está en la línea de proceso y los recursos, no dentro de la entidad.  
En MBA, un *agente* (por ejemplo, una persona, empresa o vehículo) es autónomo: tiene estado interno, reglas de decisión, objetivos e interacciones con otros agentes y con su entorno; actúa continuamente en el tiempo o al ritmo de sus propias reglas, generando comportamientos emergentes.

Esta diferencia hace que DES sea ideal para sistemas de flujo y capacidad (colas, procesos operativos) donde el interés es medir tiempos de espera, utilización de recursos y throughput. En cambio, MBA es adecuado para sistemas complejos con heterogeneidad y retroalimentaciones micro-macro (movilidad, propagación, mercados, redes) donde importan interacciones locales y decisiones individuales.

- **Ejemplo DES:** un centro de llamadas con llegadas de clientes, múltiples agentes telefónicos, prioridades, *SLA* y reintentos.  
- **Ejemplo MBA:** propagación de rumores o enfermedad en una ciudad con individuos que se mueven, deciden y cambian su comportamiento según su red social y percepciones de riesgo.

### Pregunta 2
**Explique por qué las colas son una consecuencia inevitable en la mayoría de los sistemas modelados con DES. ¿Qué tipo de información valiosa podemos extraer al analizar la longitud de una cola y el tiempo de espera en ella a lo largo de una simulación?**

Con llegadas y tiempos de servicio variables (incertidumbre), incluso si la utilización promedio es menor que 1, la variabilidad provoca congestión temporal: llegan “ráfagas” y los servicios tienen duraciones irregulares, así que se forman colas. En régimen estable, la teoría de colas muestra que la relación entre el número promedio en el sistema $L$, la tasa de llegada efectiva $\lambda$ y el tiempo promedio en el sistema $W$ cumple la Ley de Little: $L=\lambda W$. Cuanto mayor es la utilización y la variabilidad (de llegadas y servicio), mayor será la cola y el tiempo de espera esperados.

Al analizar longitud de cola y tiempos de espera (promedios, percentiles y colas de distribución) puedo:
- Detectar cuellos de botella y puntos de saturación.  
- Dimensionar recursos (servidores, puestos, inventario) para cumplir SLA.  
- Evaluar políticas (prioridades, citas, *pooling* de recursos, *batching*).  
- Cuantificar riesgo operativo (picos, *tail risk*).  
- Proponer trade-offs costo–servicio (por ejemplo, cuántos servidores adicionales reducen el $95^\text{o}$ percentil de espera a un objetivo).

### Pregunta 3
**En clase se mencionaron los recursos móviles, estáticos y portátiles. Describa un escenario (diferente al hospital) que involucre los tres tipos de recursos. Identifique qué elementos serían cada tipo de recurso y cómo interactuarían.**

Centro de distribución de e‑commerce:
- **Recursos estáticos:** estaciones de picking y empaque fijas, y cintas transportadoras en ubicaciones específicas.  
- **Recursos móviles:** robots AMR o montacargas que se desplazan entre racks, estaciones y muelles. Son solicitados para transportar pedidos o contenedores.  
- **Recursos portátiles:** totes (contenedores) y escáneres de mano que acompañan al pedido mientras avanza por el proceso; se “apoderan” (seize) al inicio y se devuelven (release) al terminar.

**Interacción:** una orden llega, se apodera de un tote (portátil) y solicita un robot (móvil) que la lleva a la estación de picking (estática). Tras el picking, otro traslado móvil la lleva a empaque (estático); el tote permanece con la orden hasta el despacho y luego retorna al pool de portátiles. 

### Pregunta 4
**Imagine que está modelando la propagación de una enfermedad en una ciudad (un problema clásico de MBA). Ahora, quieres añadir el proceso de vacunación en centros de salud. ¿Cómo podría combinar MBA y DES? ¿Qué parte del modelo sería MBA y cuál DES? ¿Qué información pasaría de un modelo a otro?**

Construiría un modelo híbrido:
- **Parte MBA (ciudad):** agentes personas con atributos (edad, estado $S/E/I/R/V$, aceptación vacunal), movilidad diaria y red de contactos. La transmisión ocurre por interacciones y decisiones individuales (por ejemplo, acudir o no a vacunarse).  
- **Parte DES (centros de salud):** flujo de pacientes por recepción → vacunación → observación, con recursos (personal, cabinas, dosis disponibles), colas, tiempos de servicio y horarios. Se mide throughput, esperas y utilización.

Intercambio de información:  
- **MBA a DES:** listas (o tasas) de llegadas por centro y franja horaria (derivadas de decisiones/ubicación de agentes), no‑shows, prioridad por riesgo.  
- **DES a MBA:** eventos de vacunación confirmados por individuo y tiempo (cambian el estado del agente a $V$ tras un delay de efectividad), tiempos de espera (que pueden afectar comportamiento futuro), y capacidad/stock (que condiciona citas futuras).  
El acople puede ser sincrónico (timestep común) o por eventos (el DES dispara eventos de vacunación que el MBA consume). Así preservo la dinámica micro de contagios (MBA) y el rendimiento operativo de la vacunación (DES).

**Referencias:** 

- GeeksforGeeks. (2020, November 19). Basics of Discrete Event Simulation using SimPy. GeeksforGeeks. https://www.geeksforgeeks.org/python/basics-of-discrete-event-simulation-using-simpy/ 
- GeeksforGeeks. (2022, July 29). Basics of NS2 and Otcl/tcl script. GeeksforGeeks. https://www.geeksforgeeks.org/computer-networks/basics-of-ns2-and-otcltcl-script/ 
- GeeksforGeeks. (2024, May 27). Queuing theory. GeeksforGeeks. https://www.geeksforgeeks.org/maths/queuing-theory/ 
- GeeksforGeeks. (2025a, April 9). Queuing models in operating system. GeeksforGeeks. https://www.geeksforgeeks.org/operating-systems/queuing-models-in-operating-system/ 
- GeeksforGeeks. (2025b, July 23). Agent based modelling. GeeksforGeeks. https://www.geeksforgeeks.org/artificial-intelligence/agent-based-modelling/ 
- GeeksforGeeks. (2025c, July 23). Simulation using R programming. GeeksforGeeks. https://www.geeksforgeeks.org/machine-learning/simulation-using-r-programming/ 
- GeeksforGeeks. (2025d, August 14). Agents in AI. GeeksforGeeks. https://www.geeksforgeeks.org/artificial-intelligence/agents-artificial-intelligence/

## Práctica


#### Inicialización del entorno y parámetros

In [2]:
import simpy
import random
import statistics

NUM_RECEPCIONISTAS = 1
NUM_MEDICOS = 2
TIEMPO_REGISTRO_PROMEDIO = 2.0      
TIEMPO_CONSULTA_PROMEDIO = 7.0      
TASA_LLEGADA_PACIENTES = 5.0        
TIEMPO_SIMULACION = 120.0          

# Lista para almacenar los tiempos totales de permanencia
tiempos_de_espera_totales = []


#### Función `paciente`: Define el proceso completo de un paciente

**Generador** que modela el ciclo de vida de un paciente en la clínica:
1. Registra su tiempo de llegada
2. Solicita un recepcionista 
3. Realiza el registro 
4. Solicita un médico 
5. Realiza la consulta 
6. Calcula y guarda el tiempo total en la clínica

In [3]:
def paciente(env, nombre, recepcionistas, medicos):   
    tiempo_de_llegada = env.now
    print(f"{nombre} llega a la clínica en el minuto {tiempo_de_llegada:.2f}")
    
    #  REGISTRO
    with recepcionistas.request() as request:
        yield request
        
        print(f"{nombre} comienza el registro en el minuto {env.now:.2f}")
        
        
        tiempo_registro = random.expovariate(1.0 / TIEMPO_REGISTRO_PROMEDIO)
        yield env.timeout(tiempo_registro)
        
        print(f"{nombre} termina el registro en el minuto {env.now:.2f}")
    
    # CONSULTA
    with medicos.request() as request:
        yield request
        
        print(f"{nombre} comienza la consulta en el minuto {env.now:.2f}")
        
        tiempo_consulta = random.expovariate(1.0 / TIEMPO_CONSULTA_PROMEDIO)
        yield env.timeout(tiempo_consulta)
        
        print(f"{nombre} termina la consulta en el minuto {env.now:.2f}")
    
    tiempo_total_en_clinica = env.now - tiempo_de_llegada
    tiempos_de_espera_totales.append(tiempo_total_en_clinica)
    
    print(f"{nombre} sale de la clínica después de {tiempo_total_en_clinica:.2f} minutos\n")

#### Función generadora de llegadas de pacientes

Esta función crea nuevos pacientes a intervalos aleatorios:
- Usa un proceso de **Poisson** para modelar llegadas 
- `random.expovariate(1/λ)` genera tiempos entre llegadas exponenciales
- Cada llegada inicia un nuevo proceso `paciente`

In [4]:
def generador_pacientes(env, recepcionistas, medicos):
   
    contador_pacientes = 0
    
    while True:
       
        tiempo_entre_llegadas = random.expovariate(1.0 / TASA_LLEGADA_PACIENTES)
        yield env.timeout(tiempo_entre_llegadas)
        
        contador_pacientes += 1
        nombre_paciente = f"Paciente {contador_pacientes}"
        
        env.process(paciente(env, nombre_paciente, recepcionistas, medicos))

#### Ejecución de la Simulación

1. Crea el entorno de SimPy
2. Define los recursos 
3. Inicia el generador de pacientes
4. Corre la simulación por el tiempo especificado

In [8]:
tiempos_de_espera_totales.clear()

env = simpy.Environment()

recepcionistas = simpy.Resource(env, capacity=NUM_RECEPCIONISTAS)
medicos = simpy.Resource(env, capacity=NUM_MEDICOS)

env.process(generador_pacientes(env, recepcionistas, medicos))


print(f" - Recepcionistas: {NUM_RECEPCIONISTAS}")
print(f" - Médicos: {NUM_MEDICOS}")
print(f" - Tiempo promedio de registro: {TIEMPO_REGISTRO_PROMEDIO} min")
print(f" - Tiempo promedio de consulta: {TIEMPO_CONSULTA_PROMEDIO} min")
print(f" - Tasa de llegada: 1 paciente cada {TASA_LLEGADA_PACIENTES} min")
print(f" - Duración de simulación: {TIEMPO_SIMULACION} min")
print(f"\n")

env.run(until=TIEMPO_SIMULACION)



 - Recepcionistas: 1
 - Médicos: 2
 - Tiempo promedio de registro: 2.0 min
 - Tiempo promedio de consulta: 7.0 min
 - Tasa de llegada: 1 paciente cada 5.0 min
 - Duración de simulación: 120.0 min


Paciente 1 llega a la clínica en el minuto 1.10
Paciente 1 comienza el registro en el minuto 1.10
Paciente 1 termina el registro en el minuto 2.71
Paciente 1 comienza la consulta en el minuto 2.71
Paciente 2 llega a la clínica en el minuto 9.16
Paciente 2 comienza el registro en el minuto 9.16
Paciente 3 llega a la clínica en el minuto 10.40
Paciente 2 termina el registro en el minuto 10.46
Paciente 2 comienza la consulta en el minuto 10.46
Paciente 3 comienza el registro en el minuto 10.46
Paciente 3 termina el registro en el minuto 10.63
Paciente 1 termina la consulta en el minuto 10.82
Paciente 1 sale de la clínica después de 9.72 minutos

Paciente 3 comienza la consulta en el minuto 10.82
Paciente 2 termina la consulta en el minuto 13.39
Paciente 2 sale de la clínica después de 4.23 minu

#### Análisis de Resultados

Se analizan las métricas clave de rendimiento:
- **Número de pacientes atendidos**: cuántos completaron el ciclo
- **Tiempo promedio en clínica**: métrica clave de experiencia del paciente
- **Tiempo máximo**: identifica cuellos de botella

In [13]:
if len(tiempos_de_espera_totales) > 0:
    num_pacientes = len(tiempos_de_espera_totales)
    tiempo_promedio = statistics.mean(tiempos_de_espera_totales)
    tiempo_maximo = max(tiempos_de_espera_totales)
    tiempo_minimo = min(tiempos_de_espera_totales)
    
    if num_pacientes >= 2:
        tiempo_mediana = statistics.median(tiempos_de_espera_totales)
        tiempos_ordenados = sorted(tiempos_de_espera_totales)
        percentil_75 = tiempos_ordenados[int(0.75 * num_pacientes)]
        percentil_95 = tiempos_ordenados[int(0.95 * num_pacientes)]
    
    print(f"\n Pacientes Procesados: {num_pacientes}")
    print(f"\n  Tiempos en la Clínica (minutos):")
    print(f"  • Promedio: {tiempo_promedio:.2f} min")
    print(f"  • Mínimo:   {tiempo_minimo:.2f} min")
    print(f"  • Máximo:   {tiempo_maximo:.2f} min")

    if num_pacientes >= 2:
        print(f"  • Mediana:  {tiempo_mediana:.2f} min")
        print(f"  • P75:      {percentil_75:.2f} min")
        print(f"  • P95:      {percentil_95:.2f} min")

    print(f"\n Análisis de Capacidad:")
    tiempo_servicio_esperado = TIEMPO_REGISTRO_PROMEDIO + TIEMPO_CONSULTA_PROMEDIO
    tasa_llegada = 1.0 / TASA_LLEGADA_PACIENTES
    
    util_recepcionistas = (tasa_llegada * TIEMPO_REGISTRO_PROMEDIO) / NUM_RECEPCIONISTAS
    util_medicos = (tasa_llegada * TIEMPO_CONSULTA_PROMEDIO) / NUM_MEDICOS
    
    print(f"   • Utilización aproximada de recepcionistas: {util_recepcionistas*100:.1f}%")
    print(f"   • Utilización aproximada de médicos: {util_medicos*100:.1f}%")
    
    if util_recepcionistas > util_medicos:
        print(f"     CUELLO DE BOTELLA: Recepción")
    elif util_medicos > util_recepcionistas:
        print(f"     CUELLO DE BOTELLA: Consulta médica")
    else:
        print(f"    Sistema balanceado")
    
else:
    print("\n  No se procesaron pacientes durante la simulación.")


 Pacientes Procesados: 22

  Tiempos en la Clínica (minutos):
  • Promedio: 10.11 min
  • Mínimo:   1.46 min
  • Máximo:   38.47 min
  • Mediana:  8.38 min
  • P75:      10.52 min
  • P95:      28.48 min

 Análisis de Capacidad:
   • Utilización aproximada de recepcionistas: 40.0%
   • Utilización aproximada de médicos: 70.0%
     CUELLO DE BOTELLA: Consulta médica


###  Un_RECEPCIONISTAS = 1, NUM_MEDICOS = 2
- 1 Recepcionista, 2 Médicos
Ejecuta la simulación con parámetros iniciales
Muestra la traza completa de eventos en consola
Calcula métricas: tiempo promedio, mínimo, máximo y mediana

In [15]:

NUM_RECEPCIONISTAS = 1
NUM_MEDICOS = 2
TIEMPO_SIMULACION = 120.0

tiempos_de_espera_totales.clear()

env = simpy.Environment()
recepcionistas = simpy.Resource(env, capacity=NUM_RECEPCIONISTAS)
medicos = simpy.Resource(env, capacity=NUM_MEDICOS)
env.process(generador_pacientes(env, recepcionistas, medicos))


print(f"Recepcionistas: {NUM_RECEPCIONISTAS}")
print(f"Médicos: {NUM_MEDICOS}")
print(f"Duración: {TIEMPO_SIMULACION} minutos")

env.run(until=TIEMPO_SIMULACION)


if len(tiempos_de_espera_totales) > 0:
    print(f"Pacientes procesados: {len(tiempos_de_espera_totales)}")
    print(f"Tiempo promedio en clínica: {statistics.mean(tiempos_de_espera_totales):.2f} minutos")
    print(f"Tiempo mínimo: {min(tiempos_de_espera_totales):.2f} minutos")
    print(f"Tiempo máximo: {max(tiempos_de_espera_totales):.2f} minutos")
    print(f"Mediana: {statistics.median(tiempos_de_espera_totales):.2f} minutos")


Recepcionistas: 1
Médicos: 2
Duración: 120.0 minutos
Paciente 1 llega a la clínica en el minuto 0.80
Paciente 1 comienza el registro en el minuto 0.80
Paciente 1 termina el registro en el minuto 1.31
Paciente 1 comienza la consulta en el minuto 1.31
Paciente 2 llega a la clínica en el minuto 1.87
Paciente 2 comienza el registro en el minuto 1.87
Paciente 1 termina la consulta en el minuto 3.06
Paciente 1 sale de la clínica después de 2.27 minutos

Paciente 2 termina el registro en el minuto 3.32
Paciente 2 comienza la consulta en el minuto 3.32
Paciente 3 llega a la clínica en el minuto 4.55
Paciente 3 comienza el registro en el minuto 4.55
Paciente 3 termina el registro en el minuto 5.39
Paciente 3 comienza la consulta en el minuto 5.39
Paciente 3 termina la consulta en el minuto 7.72
Paciente 3 sale de la clínica después de 3.17 minutos

Paciente 4 llega a la clínica en el minuto 10.95
Paciente 4 comienza el registro en el minuto 10.95
Paciente 4 termina el registro en el minuto 11.0

- ***¿En qué etapa del proceso (registro o consulta) parece que los pacientes invierten la mayor parte de su
tiempo de espera?***
La mayor parte del tiempo de espera ocurre en la etapa de consulta médica.
- Se observan varios casos donde un paciente inicia la consulta varios minutos después de haber terminado el registro como por ejemplo, Paciente 4 espera desde el minuto 11 hasta 47 para terminar la consulta.
- En cambio, los tiempos de registro son breves 
y rara vez se acumulan más de uno o dos pacientes esperando recepción.

En terminos de la utilización: 
$$U_{recepcion} = \frac{2.0}{50*1} = 0.4 (40\%)$$
$$U_{medicos} = \frac{7}{5*2} = 0.7 (70\%)$$

La consulta termina siendo el cuello de botella del sistema, ya que la utilización de los médicos es significativamente mayor que la de los recepcionistas.

- ***¿Cuál es el tiempo promedio total que un paciente permanece en la clínica según este
escenario base?***

Tiempo promedio total en la clínica

Del resultado final:

$$Promedio=10.04 minutos$$

con un rango entre 2.27 min (mínimo) y 36.57 min (máximo),
y una mediana de 8.90 minutos.

### Experimento 2: Incremento de Médicos (1 Recepcionista, 5 Médicos)

Aumentar significativamente el número de médicos para identificar el cuello de botella.

In [16]:

NUM_RECEPCIONISTAS = 1
NUM_MEDICOS = 5
TIEMPO_SIMULACION = 120.0

tiempos_de_espera_totales.clear()

env = simpy.Environment()
recepcionistas = simpy.Resource(env, capacity=NUM_RECEPCIONISTAS)
medicos = simpy.Resource(env, capacity=NUM_MEDICOS)
env.process(generador_pacientes(env, recepcionistas, medicos))

print(f"Recepcionistas: {NUM_RECEPCIONISTAS}")
print(f"Médicos: {NUM_MEDICOS}")
print(f"Duración: {TIEMPO_SIMULACION} minutos")


env.run(until=TIEMPO_SIMULACION)


if len(tiempos_de_espera_totales) > 0:
    print(f"Pacientes procesados: {len(tiempos_de_espera_totales)}")
    print(f"Tiempo promedio en clínica: {statistics.mean(tiempos_de_espera_totales):.2f} minutos")
    print(f"Tiempo mínimo: {min(tiempos_de_espera_totales):.2f} minutos")
    print(f"Tiempo máximo: {max(tiempos_de_espera_totales):.2f} minutos")
    print(f"Mediana: {statistics.median(tiempos_de_espera_totales):.2f} minutos")


Recepcionistas: 1
Médicos: 5
Duración: 120.0 minutos
Paciente 1 llega a la clínica en el minuto 3.75
Paciente 1 comienza el registro en el minuto 3.75
Paciente 1 termina el registro en el minuto 5.59
Paciente 1 comienza la consulta en el minuto 5.59
Paciente 2 llega a la clínica en el minuto 6.60
Paciente 2 comienza el registro en el minuto 6.60
Paciente 3 llega a la clínica en el minuto 6.68
Paciente 1 termina la consulta en el minuto 7.92
Paciente 1 sale de la clínica después de 4.17 minutos

Paciente 4 llega a la clínica en el minuto 8.00
Paciente 2 termina el registro en el minuto 8.67
Paciente 2 comienza la consulta en el minuto 8.67
Paciente 3 comienza el registro en el minuto 8.67
Paciente 3 termina el registro en el minuto 8.72
Paciente 3 comienza la consulta en el minuto 8.72
Paciente 4 comienza el registro en el minuto 8.72
Paciente 5 llega a la clínica en el minuto 9.34
Paciente 4 termina el registro en el minuto 11.22
Paciente 4 comienza la consulta en el minuto 11.22
Pacie

***A continuación, usted deberá investigar cuál es el factor limitante principal del sistema. Mantenga el
número de recepcionistas en NUM_RECEPCIONISTAS = 1 y realice un cambio significativo en el número de
médicos, aumentándolo a NUM_MEDICOS = 5***

**a.** Al aumentar los médicos de 2 a 5, no se observa una disminución sustancial en el tiempo promedio total de los pacientes:  
- Pasa de 10.04 min → 10.08 min, lo que evidencia casi ningún cambio en el rendimiento global.

**b.** Con base en este resultado, el recurso limitante del sistema es la recepción.  
- Aunque hay más médicos disponibles, los pacientes siguen entrando lentamente al sistema porque solo hay un recepcionista registrando a uno a la vez.  
- Los médicos permanecen parcialmente ociosos mientras los pacientes esperan ser registrados.



### Experimento 3: Incremento de Recepcionistas (2 Recepcionistas, 2 Médicos)

Restablecer médicos a 2 y aumentar recepcionistas para atacar el cuello de botella identificado.

In [18]:

NUM_RECEPCIONISTAS = 2
NUM_MEDICOS = 2
TIEMPO_SIMULACION = 120.0

tiempos_de_espera_totales.clear()

env = simpy.Environment()
recepcionistas = simpy.Resource(env, capacity=NUM_RECEPCIONISTAS)
medicos = simpy.Resource(env, capacity=NUM_MEDICOS)
env.process(generador_pacientes(env, recepcionistas, medicos))


print(f"Recepcionistas: {NUM_RECEPCIONISTAS}")
print(f"Médicos: {NUM_MEDICOS}")
print(f"Duración: {TIEMPO_SIMULACION} minutos")


env.run(until=TIEMPO_SIMULACION)


if len(tiempos_de_espera_totales) > 0:
    print(f"Pacientes procesados: {len(tiempos_de_espera_totales)}")
    print(f"Tiempo promedio en clínica: {statistics.mean(tiempos_de_espera_totales):.2f} minutos")
    print(f"Tiempo mínimo: {min(tiempos_de_espera_totales):.2f} minutos")
    print(f"Tiempo máximo: {max(tiempos_de_espera_totales):.2f} minutos")
    print(f"Mediana: {statistics.median(tiempos_de_espera_totales):.2f} minutos")
print("="*70)

Recepcionistas: 2
Médicos: 2
Duración: 120.0 minutos
Paciente 1 llega a la clínica en el minuto 4.29
Paciente 1 comienza el registro en el minuto 4.29
Paciente 1 termina el registro en el minuto 5.87
Paciente 1 comienza la consulta en el minuto 5.87
Paciente 1 termina la consulta en el minuto 6.62
Paciente 1 sale de la clínica después de 2.33 minutos

Paciente 2 llega a la clínica en el minuto 9.57
Paciente 2 comienza el registro en el minuto 9.57
Paciente 2 termina el registro en el minuto 11.08
Paciente 2 comienza la consulta en el minuto 11.08
Paciente 2 termina la consulta en el minuto 12.73
Paciente 2 sale de la clínica después de 3.16 minutos

Paciente 3 llega a la clínica en el minuto 16.22
Paciente 3 comienza el registro en el minuto 16.22
Paciente 3 termina el registro en el minuto 17.37
Paciente 3 comienza la consulta en el minuto 17.37
Paciente 4 llega a la clínica en el minuto 19.63
Paciente 4 comienza el registro en el minuto 19.63
Paciente 4 termina el registro en el minu


**Parámetros de esta corrida:** 2R–2M, 120 min  
**Resultados reportados:**  
- Pacientes procesados: 22  
- **Promedio:** **9.20 min**  
- Mín: 2.33 — Máx: 26.39 — Mediana: 8.11

#### a) Comparación con la línea base
- **Línea base (1R–2M):** 10.04 min  
- **Este escenario (2R–2M):** **9.20 min**  
**Cambio:** −0.84 min (**≈ −8.4%**). 



#### b) Intervención más efectiva
- **Aumentar médicos (2→5)** con 1 recepcionista: efecto casi nulo en el promedio. (10.04 → 10.08 min).  
- **Aumentar recepcionistas (1→2)** con 2 médicos se ve una mejora clara ~10.04 → **9.20** , representando una diferencia de −8.4%.

**Conclusión:** 

En estas condiciones de demanda (λ ≈ 1/5 min), aumentar recepcionistas fue la palanca más efectiva para reducir el tiempo total. La consulta ya operaba sin colas sistemáticas; por eso agregar médicos no aportó mejora relevante, mientras que duplicar registro suaviza la entrada y reduce las esperas residuales antes de consultar.

En las 3 simulaciones, el aumento en el número de médicos no produjo una mejora significativa en el tiempo promedio total que un paciente permanece en la clínica. Esto se debe a que el cuello de botella principal del sistema es la recepción, donde los pacientes experimentan la mayor parte de su tiempo de espera. Al aumentar el número de recepcionistas, se logró una reducción más notable en el tiempo promedio total, ya que se agilizó el proceso de registro y se redujeron las esperas antes de la consulta médica.


Se recomienda a la dirección de la clínica priorizar la ampliación del personal de recepción, incrementando de uno a dos recepcionistas, manteniendo el número actual de médicos.  
Esta medida:
- Reduce los tiempos de espera promedio en aproximadamente un 8 %.  
- Mejora la utilización de los médicos existentes, evitando periodos de inactividad.  
- Optimiza el flujo de pacientes desde el ingreso hasta la salida, sin requerir una inversión sustancial.  


### Uso de Gen AI - Documentación

#### Prompt Utilizado:
```
Ayúdame a construir una simulación de eventos discretos paso a paso en SimPy para modelar 
una clínica con dos etapas (registro y consulta). Necesito entender cada concepto:

1. Explica cómo funcionan los generadores en Python y por qué SimPy los usa
2. Muestra cómo implementar la función paciente con requests de recursos
3. Explica la diferencia entre distribuciones exponenciales y uniformes para tiempos
4. Ayúdame a crear el generador de pacientes usando proceso Poisson
5. Explica cómo calcular métricas de utilización y detectar cuellos de botella

Quiero implementarlo yo mismo pero entendiendo la teoría detrás.
```

#### ¿Por qué funcionó este prompt?

1. **Solicita comprensión, no solo código**: Pedí explicaciones de conceptos (generadores, distribuciones, Poisson)
2. **Enfoque paso a paso**: Dividí el problema en partes manejables
3. **Contexto específico**: Mencioné SimPy, el problema de la clínica, y las métricas que necesito
4. **Énfasis en aprendizaje**: Dejé claro que quiero implementar y entender, no solo copiar
